### PR TITLE
Add support for podman quadlets

### PR DESCRIPTION
**Pull request by dabelenda**
#### Pull Request (PR) description
This PR adds support for podman Quadlets, a generator for systemd that creates units automatically for files in a specific folder.

Tested with a kube Yaml file and it starts the pod.

### CODE REVIEW

File: manifests/quadlet_file.pp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#
+# @param service_restart
+#   restart (notify) the service when unit file changed
+define systemd::quadlet_file (
+  Enum['present', 'absent']                $ensure    = 'present',
+  Stdlib::Absolutepath                     $path      = '/etc/containers/systemd',
+  Optional[Variant[String, Sensitive[String], Deferred]] $content = undef,
+  Optional[String]                         $source    = undef,
+  String                                   $owner     = 'root',
+  String                                   $group     = 'root',
```
Comment (bastelfreak) [inline manifests/quadlet_file.pp:63]: can you please update those so we don't allow empty string? `String[1]`

Comment (dabelenda) [reply manifests/quadlet_file.pp]: Done.
Note: it was copy-pasted from `unit_file.pp`

File: manifests/quadlet_file.pp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+#   restart (notify) the service when unit file changed
+define systemd::quadlet_file (
+  Enum['present', 'absent']                $ensure    = 'present',
+  Stdlib::Absolutepath                     $path      = '/etc/containers/systemd',
+  Optional[Variant[String, Sensitive[String], Deferred]] $content = undef,
+  Optional[String]                         $source    = undef,
+  String[1]                                $owner     = 'root',
+  String[1]                                $group     = 'root',
+  String[1]                                $mode      = '0444',
+  Optional[Boolean]                        $enable    = undef,
```
Comment (bastelfreak) [inline manifests/quadlet_file.pp:65]: do those booleans need to be undef or can we default to true or false?

Comment (dabelenda) [reply manifests/quadlet_file.pp:65]: Not mandatory, it was copy/pasted directly from `unit_file.pp` to keep a consistent API in the module.

If you request a change, I can remove the Optional, and set a default value.

**Comment by traylenator** [comment]
I'd been thinking about doing this.

If this was going to be done I would at least steal the types I added in https://github.com/southalc/podman/pull/83 so the contents of the quadlet files can be managed.

But I question if managing quadlets should be in here at all. These are not unit files processed by systemd. Having said that its easy place to have them and it will not intrude into systemd configuration.

I think a seperate quadlet module using as much of this module as possible would make more sense? Which I'm very happy to work on lifting much of what I did above. Result is somewhat we ignoring the existing podman module but that is fine for the sake of actual progress.

**Comment by dabelenda** [comment]
@traylenator at least for unit_files this module does not manage the content of the files.

All type definitions and templating were created for the dropins. I copied the behavious of the unit_file.pp for everything which is why I did not add types.

I am also unsure if this is the correct spot for this, which is why I did not go all the way, like creating tests and managing the content of the quadlet files.

**Comment by ekohl** [comment]
Like @traylenator I question it. It's not systemd, but podman. And even with podman you need a new version (4.4+ for basic support, which Debian 12 doesn't even have, newer for the more advanced features).

Given the lack of progress on https://github.com/southalc/podman/pull/83 I wouldn't mind taking over that module. Ideally transferring but forking if needed.

**Comment by ekohl** [comment]
I currently don't use the podman module, but I can see how quadlets replace many of the old ways. A puppet-quadlets module sounds like a good plan to me.

**Comment by TheMeier** [comment]
To me it is also not really clear to which domains really belong. Podman? Systemd? I tend more towards podman, thats also where its code is now.

**Comment by dabelenda** [comment]
I agree this is a bit fuzzy in terms of scope, the generator for quadlet is in the podman package. At the same time, there are generators for a bunch of "core" features of podman, like mount points, etc.

An additionnal puppet module for quadlets that installs podman, manages the content of quadlet files, might be a good thing.
Will the new module depend on puppet/systemd for management of daemon-reload or things like that? Will the module contain its own logic for managing service like puppet/systemd ?

Conceptually the two modues will be pretty similar in features I think.

**Comment by TheMeier** [comment]
If there is going to be a new module, I think it should re-use whatever it can from the systemd module

**Comment by traylenator** [comment]
Add quadlet support in own module. - https://github.com/voxpupuli/puppet-quadlets/issues/1

**Comment by dabelenda** [comment]
Closing this PR since we agree that going for another module is the correct path.

**Pull request closed without merging**